### PR TITLE
Week 3 homework muhammad hussein

### PR DESCRIPTION
**Pull request by MA-Hussein**

### CODE REVIEW

File: Week3/homework/1-compliment.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ function givecomplements (name){
+
+    const compliments=["great", "awesome","trusty","respectful", "friendly", "clever", "Generous", "brilliant", "handsome", "beautiful"];
+
+   return `You are ${compliments[Math.floor(Math.random() * compliments.length)]}, ${name}`+"!";
+}
+
+console.log(givecomplements("Muhammad"));
+console.log(givecomplements("Muhammad"));
+console.log(givecomplements("Muhammad"));
```
Comment (nudge) [inline Week3/homework/1-compliment.js:12]: 👍

File: Week3/homework/3-fortuneTeller.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict'
+
+function random(array){
+  return Math.floor(Math.random() * array.length);
+}
```
Comment (nudge) [inline Week3/homework/3-fortuneTeller.js:5]: Nice use of another function here!

File: Week3/homework/3-fortuneTeller.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+const partners = ["Mariam","Lura","Paula", "Sophia","karlijn"];
+const locations = [ "Amsterdam", "Paris", "new york", "Tokyo","Berlin"];
+const jobs = ["Developer", "Actor", "Doctor", "Lawyer", "Accountant"];
+
+function tellFortune(children, partners, locations, jobs){
+  return `You will be a ${jobs[random(jobs)]} in ${locations[random(locations)]}, and married to ${partners[random(partners)]} with ${children[random(children)]} kids.`
+}
+
+let fortune = tellFortune(children, partners, locations, jobs);
+console.log(fortune);
```
Comment (nudge) [inline Week3/homework/3-fortuneTeller.js:17]: 👍

File: Week3/homework/4-shoppingCart.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+function addToShoppingCart(item) {
+    groceryItems.push(item);
+    if (groceryItems.length > 3) {
+        groceryItems.shift()
+    }
+    return "You bought " + groceryItems.join(" ") + "!";
+};
+console.log(addToShoppingCart("bread"));
+console.log(addToShoppingCart("orange"));
+console.log(addToShoppingCart("Tomato"));
```
Comment (nudge) [inline Week3/homework/4-shoppingCart.js:13]: 👍

File: Week3/homework/5-totalPriceCalculator.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ 
+  function calculateTotalPrice(price) {
+    let sum = 0;
+    for (let key in price) {
+      sum += price[key];
+    }
+    return sum;
+  }
+  
+  console.log(calculateTotalPrice(cartForParty));
```
Comment (nudge) [inline Week3/homework/5-totalPriceCalculator.js:20]: 👍

File: Week3/homework/2-dogyears.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+'use strict'
+
+function DogAgeCulculator(DogAge){
+	
+  return `Your dog is ${DogAge * 7} years old in dog years!`;
+}
+
+console.log(DogAgeCulculator(2));
+console.log(DogAgeCulculator(4));
+console.log(DogAgeCulculator(6));
```
Comment (nudge) [inline Week3/homework/2-dogyears.js:11]: 👍

File: Week3/homework/creditCardValidator.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+ function sum_of_digits_Validator(creditCardNumber) {
+  for (const digits of creditCardNumber) {
+    sum_of_digits += +digits;
+  }
+  return sum_of_digits;
+ }
+ //This function checks whether there is at least two different digits(all of the digits cannot be the same)
+	
+ function allSameDigitsValidator(creditCardNumber) {
+        return /^(\d)\1+$/.test(creditCardNumber.replace(/[^\d]/g,''));
```
Comment (nudge) [inline Week3/homework/creditCardValidator.js:18]: funky use of regular expressions!

File: Week3/homework/creditCardValidator.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        return /^(\d)\1+$/.test(creditCardNumber.replace(/[^\d]/g,'')); 
+    }
+	
+	
+	/* This function checks whether the count of digits is more than 16 .    and makes sure that all of them are numbers and the last digit is      even ,it calles also the previous functions. 
+	*/
+
+    if (creditCardNumber.toString().length > 16 || creditCardNumber.toString().length < 16) { //This if statement checks  the credit card number whether all.
+        return 'Invalid number:Your credit card namber must be at least 16 digit.'
+    }  else if (isNaN(creditCardNumber)) {
```
Comment (nudge) [inline Week3/homework/creditCardValidator.js:27]: nice use of `isNaN()`

File: Week3/homework/creditCardValidator.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        return 'Invalid number:Your credit card number should consist of numbers only.'
+    } else if (creditCardNumber.toString()[15] % 2 !== 0) { 
+        return 'Invalid number:The final digit must be even.'
+    } else if (sum_of_digits_Validator(creditCardNumber) < 16) {
+        return 'Invalid number:The sum of all the digits must be greater than 16.'
+    } else if (allSameDigitsValidator(creditCardNumber) === true) {
```
Comment (nudge) [inline Week3/homework/creditCardValidator.js:33]: Good use of functions to check these validations. Nit: try to keep the naming the same style, e.g. all `camelCase` or `snake_case`.

File: Week3/homework/creditCardValidator.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+}
+//Valid numbers
+console.log(creditCardValidator('9999777788880000'));
+console.log(creditCardValidator('6666666666661666'));
+//Invalid numbers
+console.log(creditCardValidator('a92332119c011112'));
+console.log(creditCardValidator('4444444444444444'));
+console.log(creditCardValidator('1111111111111110'));
+console.log(creditCardValidator('6666666666666661'));
```
Comment (nudge) [inline Week3/homework/creditCardValidator.js:47]: 👍

**Review by nudge — COMMENTED** [review]
Next time don't forget to check out `master` before creating a new branch for homework, this separates each homework so they don't all appear in one pull request.

**Comment by MA-Hussein** [comment]
> Next time don't forget to check out `master` before creating a new branch for homework, this separates each homework so they don't all appear in one pull request.

I will do , thanks for the review  David @nudge .

**Pull request closed without merging**